src/sharing/components/ShareByEmail.jsx

```diff
@@ -225,13 +150,30 @@ class ShareByEmail extends Component {
   }
 
   onRecipientPick = recipient => {
-    const existing = this.state.recipients.find(r => r === recipient)
-    if (!existing) {
-      this.setState(state => ({
-        ...state,
-        recipients: [...state.recipients, recipient]
-      }))
+    let contactsToAdd
+    if (recipient._type === DOCTYPE_GROUPS) {
+      const groupId = recipient.id
+      contactsToAdd = this.props.contacts.data.filter(contact => {
+        const contactGroupIds = get(
+          contact,
+          'relationships.groups.data',
+          []
+        ).map(group => group._id)
+
+        return contactGroupIds.includes(groupId)
+      })
+    } else {
+      contactsToAdd = [recipient]
     }
+
+    const filtered = contactsToAdd
+      .filter(contact => contact.email.length > 0)
```

Comment:
So if the contact has no email, we don't add it?

Reply:
Good catch. What do we want = If it has no email or cozy we don't add it (so I'll fix the cozy part)

Reply:
Hum, the contact is rejected by the stack if it has no email: `Unhandled promise rejection FetchError: "{"errors":[{"status":"500","title":"Unqualified error","detail":"The contact has no email address","source":{}}]}"`

Reply:
(created a card on trello)